src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -458,5 +458,60 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
 
     @. ᶜu⁰ = C123(Y.c.uₕ) + ᶜinterp(C123(ᶠu³⁰))
 
+    @. ᶜlinear_buoygrad = buoyancy_gradients(
+        params,
+        moisture_model,
+        EnvBuoyGrad(
+            BuoyGradMean(),
+            TD.air_temperature(thermo_params, ᶜts),                           # t_sat
+            TD.vapor_specific_humidity(thermo_params, ᶜts),                   # qv_sat
+            q_tot,                                                            # qt_sat
+            TD.dry_pottemp(thermo_params, ᶜts),                               # θ_sat
+            TD.liquid_ice_pottemp(thermo_params, ᶜts),                        # θ_liq_ice_sat
+            projected_vector_data(
+                C3,
+                ᶜgradᵥ(ᶠinterp(TD.virtual_pottemp(thermo_params, ᶜts))),
+                ᶜlg,
+            ),                                                                 # ∂θv∂z_unsat
+            projected_vector_data(C3, ᶜgradᵥ(ᶠinterp(q_tot)), ᶜlg),            # ∂qt∂z_sat
+            projected_vector_data(
+                C3,
+                ᶜgradᵥ(ᶠinterp(TD.liquid_ice_pottemp(thermo_params, ᶜts))),
+                ᶜlg,
+            ),                                                                 # ∂θl∂z_sat
+            ᶜp,                                                                # p
+            ifelse(TD.has_condensate(thermo_params, ᶜts), 1, 0),               # en_cld_frac
+            Y.c.ρ,                                                             # ρ
+        ),
+    )
+    @. ᶜshear² = FT(1e-4)
+    ᶜprandtl_nvec = p.ᶜtemp_scalar
+    @. ᶜprandtl_nvec = 1
+    ᶜtke_exch = p.ᶜtemp_scalar_2
+    @. ᶜtke_exch = 0
+    for j in 1:n
+        @. ᶜtke_exch +=
+            ᶜρaʲs.:($$j) * ᶜentr_detrʲs.:($$j).detr / Y.c.ρ * (
+                1 / 2 *
+                (
+                    get_physical_w(ᶜuʲs.:($$j), ᶜlg) - get_physical_w(ᶜu⁰, ᶜlg)
+                )^2 - ᶜtke⁰
+            )
+    end
+
+    sfc_tke = Fields.level(ᶜtke⁰, 1)
+    @. ᶜmixing_length = mixing_length(
```

Comment:
This function also increases the allocation a bit (again, same for progrnostic edmf, and the jet failure doesn't change). With this function the allocation increases from 10480 to 10704. The function is here: https://github.com/CliMA/ClimaAtmos.jl/blob/657e87d1fc0220d1678489129a2ec556cc7bf3a7/src/prognostic_equations/edmfx_closures.jl#L265

Reply:
Actually maybe not. Let me check again.

Reply:
No, this function does not allocate.